src/main/java/gift/service/ProductService.java

```diff
@@ -25,38 +26,27 @@ public ProductService(ProductRepository productRepository, CategoryRepository ca
         this.categoryRepository = categoryRepository;
     }
 
-    public boolean addNewProduct(ProductDto productDto){
+    public void addNewProduct(ProductDto productDto){
         if (productRepository.existsByName(new ProductName(productDto.name()))) {
-            return false;
+            throw new ValueAlreadyExistsException("ProductName already exists in Database");
         }
         Category category = findCategory(productDto.categoryName());
-        Product product = new Product(category,new ProductName(productDto.name()),productDto.price(),productDto.imageUrl(),productDto.amount());
+        Product product = new Product(category,new ProductName(productDto.name()),productDto.price(),productDto.imageUrl());
 
         productRepository.save(product);
-        return true;
     }
 
-    public boolean updateProduct(Long id, ProductDto productDto) {
+    public void updateProduct(Long id, ProductDto productDto) {
         Optional<Product> product = productRepository.findById(id);
-        if(product.isPresent()){
-            Product updateProduct = product.get();
-            Category category = findCategory(productDto.categoryName());
-
-            Product newProduct = new Product(category,new ProductName(productDto.name()),productDto.price(),productDto.imageUrl(),productDto.amount());
-            updateProduct.updateProduct(newProduct);
-            productRepository.save(updateProduct);
-            return true;
+        if(product.isEmpty()){
+            throw new ValueNotFoundException("Product not exists in Database");
```

Comment:
optional을 아래와 같이 활용해보시는걸 추천드립니다.

```suggestion
        Product updateProduct = productRepository.findById(id)
                .orElseThrow(() -> new ValueNotFoundException("Product not exists in Database"));
```